.gitlab/docker/Dockerfile.spack_compiler

```diff
@@ -0,0 +1,32 @@
+ARG BASE_IMAGE
+FROM $BASE_IMAGE
+
+ARG ARCH
+ARG BOOST
+ARG BUILD_TYPE
+ARG COMPILER
+ARG CXXSTD
+ARG HWLOC
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+SHELL ["/bin/bash", "-c"]
+
+# Disable host compatibility to be able to compile for broadwell on zen2
+RUN echo -e "concretizer:\n  targets:\n    host_compatible: false" \
+    > $SPACK_ROOT/etc/spack/concretizer.yaml
```

Comment:
I think this can be done with `spack config add ...`. The json can also be written without newlines: `concretizer:targets:host_compatible:false`. I think the comment above is also unnecessarily specific. It'll be compiling for haswell, zen3, etc. in the future as well.

Reply:
The following at least works for me locally on a recent spack develop (see `<--`, added by me to the output):
```
~> spack config blame concretizer                                                                                                           [0/0]
---                                                         concretizer:
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:18    reuse: dependencies
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:23    targets:
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:28      granularity: microarchitectures
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:32      host_compatible: True <---
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:38    unify: True
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:40    duplicates:
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:44      strategy: none
~> spack config --scope defaults add "concretizer:targets:host_compatible:false"
~> spack config blame concretizer
---                                                         concretizer:
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:7     reuse: dependencies
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:12    targets:
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:17      granularity: microarchitectures
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:18      host_compatible: False <---
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:19    unify: True
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:21    duplicates:
/home/mjs/src/spack/etc/spack/defaults/concretizer.yaml:25      strategy: none
```